src/Mod/Sketcher/App/ExternalGeometryExtension.cpp

```diff
@@ -36,29 +36,50 @@ using namespace Sketcher;
 constexpr std::array<const char*, ExternalGeometryExtension::NumFlags>
     ExternalGeometryExtension::flag2str;
 
-TYPESYSTEM_SOURCE(Sketcher::ExternalGeometryExtension, Part::GeometryPersistenceExtension)
+TYPESYSTEM_SOURCE(Sketcher::ExternalGeometryExtension, Part::GeometryMigrationPersistenceExtension)
 
 void ExternalGeometryExtension::copyAttributes(Part::GeometryExtension* cpy) const
 {
     Part::GeometryPersistenceExtension::copyAttributes(cpy);
 
     static_cast<ExternalGeometryExtension*>(cpy)->Ref = this->Ref;
+    static_cast<ExternalGeometryExtension*>(cpy)->RefIndex = this->RefIndex;
     static_cast<ExternalGeometryExtension*>(cpy)->Flags = this->Flags;
 }
 
 void ExternalGeometryExtension::restoreAttributes(Base::XMLReader& reader)
 {
     Part::GeometryPersistenceExtension::restoreAttributes(reader);
 
-    Ref = reader.getAttribute("Ref");
-    Flags = FlagType(reader.getAttribute("Flags"));
+    Ref = reader.getAttribute("Ref", "");
+    RefIndex = reader.getAttributeAsInteger("RefIndex", "-1");
+    Flags = FlagType(reader.getAttributeAsUnsigned("Flags", "0"));
 }
 
 void ExternalGeometryExtension::saveAttributes(Base::Writer& writer) const
 {
     Part::GeometryPersistenceExtension::saveAttributes(writer);
+    // For compatibility with upstream FreeCAD, always save 'Ref' and 'Flags'.

```

Comment:
Since we *are* upstream FreeCAD, I suggest dropping this comment, and the two commented-out lines below.